ath79: add support for D-Link DIR-842 C2

programmingisart opened the pull request:
Hardware spec of DIR-842 C2:
SoC: QCA9563
DRAM: 128MB DDR2
Flash: 16MB SPI-NOR
Switch: QCA8337N
WiFi 5.8GHz: QCA9888
WiFi 2.4Ghz: QCA9563
USB: 2.0

Flash instructions:

1. Upgrade the factory.bin through the factory web interface or the u-boot
   failsafe interface.
   The firmware will boot up correctly for the first time.
   Do not power off the device after OpenWrt has booted. Otherwise the u-boot
   will enter failsafe mode as the checksum of the firmware has been changed.
2. Upgrade the sysupgrade.bin in OpenWrt.
   After upgrading completes the u-boot won't complain about the firmware
   checksum and it's OK to use now.
3. If you powered off the device before upgrading the sysupgrade.bin, just
   upgrade the factory.bin through the u-boot failsafe interface and then goto
   step 2.

Signed-off-by: Jackson Lim <jackcolentern@gmail.com>

jackcolentern commented:
The mac address printed on the label is used by LAN
eth0.1

blocktrron commented on target/linux/ath79/dts/qca9563_dlink_dir-842-c2.dts:
```diff
@@ -0,0 +1,198 @@
+		gpios = <11 GPIO_ACTIVE_LOW>;
+		line-name = "reset-button";
+	};
+
```
This newline can be removed

blocktrron commented on target/linux/ath79/dts/qca9563_dlink_dir-842-c2.dts:
```diff
@@ -0,0 +1,198 @@
+		};
+	};
+	
+	reset-button {
```
What purpose does this hog serve? Is the reset button not working without it?

jackcolentern replied:
The hog pull the gpio high on boot, reset button doesnt work without it

blocktrron replied:
I think a short comment describing the necessity prevents possible future confusion. So maybe add a short oneliner here?

jackcolentern replied:
Sure I will add a comment

blocktrron commented on target/linux/ath79/dts/qca9563_dlink_dir-842-c2.dts:
```diff
@@ -0,0 +1,198 @@
+
+	phy0: ethernet-phy@0 {
+		reg = <0>;
+		phy-mode = "sgmii";
```
Looking at the ar8327 driver, we should not need this property, PHY mode for the individual ports are set thru `qca,ar8327-initvals` property.

Can yo ureport back if it still works without the `phy-mode` here?

jackcolentern replied:
The dts wont compile after I removed that...

blocktrron replied:
That sounds strange. It should compile even in case the driver needs this property. What was the exact error? Build with `make -j1 V=s` to see the error.

jackcolentern replied:
```
ake[5]: Entering directory '/home/ha/gitopenwrt2/openwrt/target/linux/ath79/image'
make[5]: Nothing to be done for 'compile'.
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar9331_8dev_carambola2.dtb.tmp ../dts/ar9331_8dev_carambola2.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar9331_8dev_carambola2.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar9331_8dev_carambola2.dtb.tmp
rm -f /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar9331_8dev_carambola2.dtb.tmp
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7161_adtran_bsap1800-v2.dtb.tmp ../dts/ar7161_adtran_bsap1800-v2.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7161_adtran_bsap1800-v2.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7161_adtran_bsap1800-v2.dtb.tmp
rm -f /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7161_adtran_bsap1800-v2.dtb.tmp
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7161_adtran_bsap1840.dtb.tmp ../dts/ar7161_adtran_bsap1840.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7161_adtran_bsap1840.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7161_adtran_bsap1840.dtb.tmp
rm -f /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7161_adtran_bsap1840.dtb.tmp
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7161_aruba_ap-105.dtb.tmp ../dts/ar7161_aruba_ap-105.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7161_aruba_ap-105.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7161_aruba_ap-105.dtb.tmp
rm -f /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7161_aruba_ap-105.dtb.tmp
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7242_avm_fritz300e.dtb.tmp ../dts/ar7242_avm_fritz300e.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7242_avm_fritz300e.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7242_avm_fritz300e.dtb.tmp
rm -f /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7242_avm_fritz300e.dtb.tmp
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9561_avm_fritz4020.dtb.tmp ../dts/qca9561_avm_fritz4020.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9561_avm_fritz4020.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9561_avm_fritz4020.dtb.tmp
rm -f /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9561_avm_fritz4020.dtb.tmp
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7242_buffalo_bhr-4grv.dtb.tmp ../dts/ar7242_buffalo_bhr-4grv.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7242_buffalo_bhr-4grv.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7242_buffalo_bhr-4grv.dtb.tmp
rm -f /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7242_buffalo_bhr-4grv.dtb.tmp
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9557_buffalo_bhr-4grv2.dtb.tmp ../dts/qca9557_buffalo_bhr-4grv2.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9557_buffalo_bhr-4grv2.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9557_buffalo_bhr-4grv2.dtb.tmp
rm -f /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9557_buffalo_bhr-4grv2.dtb.tmp
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7161_buffalo_wzr-hp-ag300h.dtb.tmp ../dts/ar7161_buffalo_wzr-hp-ag300h.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7161_buffalo_wzr-hp-ag300h.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7161_buffalo_wzr-hp-ag300h.dtb.tmp
rm -f /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7161_buffalo_wzr-hp-ag300h.dtb.tmp
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7242_buffalo_wzr-hp-g302h-a1a0.dtb.tmp ../dts/ar7242_buffalo_wzr-hp-g302h-a1a0.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7242_buffalo_wzr-hp-g302h-a1a0.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7242_buffalo_wzr-hp-g302h-a1a0.dtb.tmp
rm -f /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7242_buffalo_wzr-hp-g302h-a1a0.dtb.tmp
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7242_buffalo_wzr-hp-g450h.dtb.tmp ../dts/ar7242_buffalo_wzr-hp-g450h.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7242_buffalo_wzr-hp-g450h.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7242_buffalo_wzr-hp-g450h.dtb.tmp
rm -f /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7242_buffalo_wzr-hp-g450h.dtb.tmp
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9533_comfast_cf-e110n-v2.dtb.tmp ../dts/qca9533_comfast_cf-e110n-v2.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9533_comfast_cf-e110n-v2.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9533_comfast_cf-e110n-v2.dtb.tmp
rm -f /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9533_comfast_cf-e110n-v2.dtb.tmp
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar9344_comfast_cf-e120a-v3.dtb.tmp ../dts/ar9344_comfast_cf-e120a-v3.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar9344_comfast_cf-e120a-v3.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar9344_comfast_cf-e120a-v3.dtb.tmp
rm -f /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar9344_comfast_cf-e120a-v3.dtb.tmp
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9531_comfast_cf-e5.dtb.tmp ../dts/qca9531_comfast_cf-e5.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9531_comfast_cf-e5.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9531_comfast_cf-e5.dtb.tmp
rm -f /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9531_comfast_cf-e5.dtb.tmp
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9558_devolo_dvl1200e.dtb.tmp ../dts/qca9558_devolo_dvl1200e.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9558_devolo_dvl1200e.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9558_devolo_dvl1200e.dtb.tmp
rm -f /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9558_devolo_dvl1200e.dtb.tmp
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9558_devolo_dvl1200i.dtb.tmp ../dts/qca9558_devolo_dvl1200i.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9558_devolo_dvl1200i.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9558_devolo_dvl1200i.dtb.tmp
rm -f /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9558_devolo_dvl1200i.dtb.tmp
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9558_devolo_dvl1750c.dtb.tmp ../dts/qca9558_devolo_dvl1750c.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9558_devolo_dvl1750c.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9558_devolo_dvl1750c.dtb.tmp
rm -f /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9558_devolo_dvl1750c.dtb.tmp
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9558_devolo_dvl1750e.dtb.tmp ../dts/qca9558_devolo_dvl1750e.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9558_devolo_dvl1750e.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9558_devolo_dvl1750e.dtb.tmp
rm -f /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9558_devolo_dvl1750e.dtb.tmp
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9558_devolo_dvl1750i.dtb.tmp ../dts/qca9558_devolo_dvl1750i.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9558_devolo_dvl1750i.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9558_devolo_dvl1750i.dtb.tmp
rm -f /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9558_devolo_dvl1750i.dtb.tmp
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9558_devolo_dvl1750x.dtb.tmp ../dts/qca9558_devolo_dvl1750x.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9558_devolo_dvl1750x.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9558_devolo_dvl1750x.dtb.tmp
rm -f /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9558_devolo_dvl1750x.dtb.tmp
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7161_dlink_dir-825-b1.dtb.tmp ../dts/ar7161_dlink_dir-825-b1.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7161_dlink_dir-825-b1.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7161_dlink_dir-825-b1.dtb.tmp
rm -f /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar7161_dlink_dir-825-b1.dtb.tmp
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar9344_dlink_dir-825-c1.dtb.tmp ../dts/ar9344_dlink_dir-825-c1.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar9344_dlink_dir-825-c1.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar9344_dlink_dir-825-c1.dtb.tmp
rm -f /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar9344_dlink_dir-825-c1.dtb.tmp
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar9344_dlink_dir-835-a1.dtb.tmp ../dts/ar9344_dlink_dir-835-a1.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar9344_dlink_dir-835-a1.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar9344_dlink_dir-835-a1.dtb.tmp
rm -f /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-ar9344_dlink_dir-835-a1.dtb.tmp
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9563_dlink_dir-859-a1.dtb.tmp ../dts/qca9563_dlink_dir-859-a1.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9563_dlink_dir-859-a1.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9563_dlink_dir-859-a1.dtb.tmp
rm -f /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9563_dlink_dir-859-a1.dtb.tmp
mips-openwrt-linux-musl-cpp -nostdinc -x assembler-with-cpp -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/arch/mips/boot/dts/include -I/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/include/ -undef -D__DTS__  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9563_dlink_dir-842-c2.dtb.tmp ../dts/qca9563_dlink_dir-842-c2.dts
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-4.19.57/scripts/dtc/dtc -O dtb -i../dts/ -Wno-unit_address_vs_reg -Wno-unit_address_vs_reg -Wno-simple_bus_reg -Wno-unit_address_format -Wno-pci_bridge -Wno-pci_device_bus_num -Wno-pci_device_reg -Wno-avoid_unnecessary_addr_size -Wno-alias_paths -Wno-graph_child_address -Wno-graph_port -Wno-unique_unit_address  -o /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9563_dlink_dir-842-c2.dtb /home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9563_dlink_dir-842-c2.dtb.tmp
/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9563_dlink_dir-842-c2.dtb: ERROR (phandle_references): /ahb/eth@19000000: Reference to non-existent node or label "phy0"

ERROR: Input tree has errors, aborting (use -f to force output)
Makefile:82: recipe for target '/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9563_dlink_dir-842-c2.dtb' failed
make[5]: *** [/home/ha/gitopenwrt2/openwrt/build_dir/target-mips_24kc_musl/linux-ath79_generic/image-qca9563_dlink_dir-842-c2.dtb] Error 2
make[5]: Leaving directory '/home/ha/gitopenwrt2/openwrt/target/linux/ath79/image'
Makefile:19: recipe for target 'install' failed
make[4]: *** [install] Error 2
make[4]: Leaving directory '/home/ha/gitopenwrt2/openwrt/target/linux/ath79'
Makefile:13: recipe for target 'install' failed
make[3]: *** [install] Error 2
make[3]: Leaving directory '/home/ha/gitopenwrt2/openwrt/target/linux'
time: target/linux/install#11.03#2.32#13.11
target/Makefile:23: recipe for target 'target/linux/install' failed
make[2]: *** [target/linux/install] Error 2
make[2]: Leaving directory '/home/ha/gitopenwrt2/openwrt'
target/Makefile:19: recipe for target '/home/ha/gitopenwrt2/openwrt/staging_dir/target-mips_24kc_musl/stamp/.target_install' failed
make[1]: *** [/home/ha/gitopenwrt2/openwrt/staging_dir/target-mips_24kc_musl/stamp/.target_install] Error 2
make[1]: Leaving directory '/home/ha/gitopenwrt2/openwrt'
/home/ha/gitopenwrt2/openwrt/include/toplevel.mk:216: recipe for target 'world' failed
make: *** [world] Error 2

```
This is the error message

blocktrron replied:
`ERROR (phandle_references): /ahb/eth@19000000: Reference to non-existent node or label "phy0"` suggests you removed more than just line 156 (`phy-mode = "sgmii";`).

The `phy0` handle is supposed to be untouched as it's referenced from the `phy-mode` property of the ethernet node (which shall be left untouched).

jackcolentern replied:
sorry of my mistake, removed it, compiled, and everything still works fine, I already committed the change

blocktrron commented on target/linux/ath79/dts/qca9563_dlink_dir-842-c2.dts:
```diff
@@ -0,0 +1,198 @@
+	#size-cells = <0>;
+	status = "okay";
+
+	hub_port0: port@1 {
```
The `hub_port0` handle can be dropped from this node as it's not referenced anywhere.

hanipouspilot commented:
The printed on label mac normally is used on wan, unless many macs are printed.

jackcolentern commented:
> The printed on label mac normally is used on wan, unless many macs are printed.

But in the devconf partition, the mac printed on label is lan mac also.

hanipouspilot commented:
The main idea with macs is to keep stock firmware behavior. So after you flash Openwrt, macs stay the same.

hanipouspilot commented:
In many cases you can find 1 or 2 macs in flash partitions. But you need to Ethernet and two wireless macs that are produced by incrementing the "base" macs some ways. There are functions in user space scripts for that.

jackcolentern commented:
> In many cases you can find 1 or 2 macs in flash partitions. But you need to Ethernet and two wireless macs that are produced by incrementing the "base" macs some ways. There are functions in user space scripts for that.

The devconf partition already has something like: lanmac=XX:XX:XX:XX:XX:XX,wanmac=YY:YY:YY:YY:YY:YY

So I think the mac printed on label is intended to be lanmac by d-link?

jackcolentern commented:
@blocktrron 
I changed most of the things you suggested, thanks!

jackcolentern commented:
Tested & working
LEDs
LAN/WAN ports
MAC address (same as defined in factory devconf partition)
USB port
5G wifi
2.4G wifi
Reset Button

programmingisart committed:
ath79: add support for D-Link DIR-842 C2

Hardware spec of DIR-842 C2:
SoC: QCA9563
DRAM: 128MB DDR2
Flash: 16MB SPI-NOR
Switch: QCA8337N
WiFi 5.8GHz: QCA9888
WiFi 2.4Ghz: QCA9563
USB: 2.0

Flash instructions:

1. Upgrade the factory.bin through the factory web interface or the u-boot
   failsafe interface.
   The firmware will boot up correctly for the first time.
   Do not power off the device after OpenWrt has booted. Otherwise the u-boot
   will enter failsafe mode as the checksum of the firmware has been changed.
2. Upgrade the sysupgrade.bin in OpenWrt.
   After upgrading completes the u-boot won't complain about the firmware
   checksum and it's OK to use now.
3. If you powered off the device before upgrading the sysupgrade.bin, just
   upgrade the factory.bin through the u-boot failsafe interface and then goto
   step 2.

Signed-off-by: Jackson Lim <jackcolentern@gmail.com>

blocktrron commented on target/linux/ath79/dts/qca9563_dlink_dir-842-c2.dts:
```diff
@@ -0,0 +1,192 @@
+&spi {
+	num-cs = <1>;
+
+	status = "okay";
```
Swap `num-cs` and `status` (and drop the newline between them).

blocktrron commented on target/linux/ath79/dts/qca9563_dlink_dir-842-c2.dts:
```diff
@@ -0,0 +1,192 @@
+				reg = <0xfe0000 0x20000>;
+				read-only;
+			};
+			
```
Drop this newline

blocktrron commented on target/linux/ath79/dts/qca9563_dlink_dir-842-c2.dts:
```diff
@@ -0,0 +1,192 @@
+
+&usb0 {
+	#address-cells = <1>;
+	#size-cells = <0>;
```
`#address-cells` and `#size-cells` are only required in case you have a subnode with a `reg` property. As this is not the case here, these two properties can be dropped

blocktrron reviewed (commented):
Hopefully, the last iteration.

Thanks for you work :+1:

jackcolentern commented:
All done, thanks!

adrianschmutzler commented:
@hanipouspilot @jackcolentern 

You were disagreeing on the label MAC. Is it lan (eth0.1) or wan (eth0.2)?

jackcolentern commented:
@adrianschmutzler It is used by eth0.1 (lan)

jackcolentern commented:
I can confirm the LAN and WAN addresses are set exactly the same as the factory D-link firmware

blocktrron commented:
LGTM.

However, i would be interested where (Country) you bought this device, as i was only able to find Information about rev. C1 & C3 here (and both lack an USB port).

jackcolentern commented:
This is the router provided by Malaysia Telco called Unifi (link: https://unifi.com.my/personal), it is simply a C1 with the USB port soldered. 

(C1 has a USB circuit on the board but the components not soldered).

BTW, this PR is tested and confirmed to work fine on C1 as well

hanipouspilot commented:
> BTW, this PR is tested and confirmed to work fine on C1 as well

Maybe it makes sense to support both devices by making a common dtsi and removing USB from C1?

blocktrron commented:
@jackcolentern But the device is explictly labled as C2?

Otherwise, I like the idea of @hanipouspilot but if this is not something you would like to do I'm fine going without C1 support here.

So add insult to injury, there seems to be a rev. C3 already.

hanipouspilot commented:
> So add insult to injury, there seems to be a rev. C3 already.
If we have the dtsi for dir-842, someone who has C3 can easily test and add it later.

jackcolentern commented:
Yes, it write "H/W Ver.: C2" on the label, it shows C2 in the original firmware as well.

How about we use this PR first, and I will try to get a C3 and find out what hardware it uses, if possible I will make a PR supporting C1 C2 C3 in one go in the future.

jackcolentern commented:
@blocktrron 
Yes, it write "H/W Ver.: C2" on the label, it shows C2 in the original firmware as well.

How about we use this PR first, and I will try to get a C3 and find out what hardware it uses, if possible I will make a PR supporting C1 C2 C3 in one go in the future (no promise though).

blocktrron commented:
Merged into my staging tree: https://git.openwrt.org/?p=openwrt/staging/blocktrron.git;a=commit;h=e9efc9154755e70dcc35926701254694868a753c
